code/create/dta_dicom2spec_rules.py

```diff
@@ -1,158 +1,115 @@
+        if image_type and "fieldmap" in protocol:
                 return "fieldmap"
 
-
-        # I deleted "t2star" from that list for this specific dataset, because it does not contain these.
-        # It would probably also be fine, if I had moved the "fieldmap" value to the start
-        direct_search_terms = ["t1", "t1w", "t2", "t2w",
-                               "t1rho", "t1map", "t2map", "flair",
-                               "flash", "pd", "pdmap", "pdt2", "inplanet1",
-                               "inplanet2", "angio", "dwi", "phasediff",
-                               "phase1", "phase2", "magnitude1", "magnitude2",
-                               "fieldmap", "epi", "meg"]
+	# this list should try to just contain values that might appear in the DTA-dataset.
+	# TODO: magnitude/phase does not appear in the dcm-headers, just fieldmap, and we correct it later?
+        direct_search_terms = ["t1", "t1w", "dwi", "phasediff", "phase1", "phase2", "magnitude1", "magnitude2", "fieldmap"]
 
         for m in direct_search_terms:
             if m in prot_parts:
                 return m
             if "mprage" in prot_parts:
                 return "t1w"
+    return None
 
-        # BEGIN: Additional rule for forrest-structural
-        # TODO: Probably to be moved to some rule enhancement
-        if "st1w" in prot_parts:
-            return "t1w"
-        if "st2w" in prot_parts:
-            return "t2w"
-        if "tof" in prot_parts:
-            return "angio"
-        # END
 
-        if "func" in prot_parts:
-            return "bold"
+def _guess_task(dcm_dict, record):
 
-    # found nothing, but modality isn't necessarily required
-    return None
+    protocol = record.get("ProtocolName", None).lower()
 
-def _guess_task(dcm_dict, record):
-    protocol = record.get("ProtocolName", None)
+    # TODO: do I really need/use this values?
     count_if_same_protocol_but_scanned_earlier = 1
-    #
     exp_flag = False
 
+    # check if protocol is unequal to None
     if protocol:
-        if protocol.lower() in "dta_epi_pmu_rest":
+        if protocol == "dta_epi_pmu_rest":
                return "rest"
-        if protocol.lower() in "DTA_ep2d_bold_Task".lower():
+        if protocol == "DTA_ep2d_bold_Task":
+
             for elem in dcm_dict._dicom_series:
-               if "DTA_ep2d_bold_Task".lower() in elem.get("ProtocolName").lower():
+               if "DTA_ep2d_bold_Task".lower() == protocol:
                   return ( "exp" + get_column( record.get("PatientID") ) )
 
-        if "60dir".lower() in protocol.lower():
+        if "60dir" in protocol:
             return "60dir"
-        if "120dir".lower() in protocol.lower():
+        if "120dir" in protocol:
             return "120dir"
 
-
         import re
-        prot_parts = re.split('_|-|\s', protocol.lower())
+        prot_parts = re.split('_|-|\s', protocol)
+
+	# TODO: is this try/except-block helpful for DTA or just for forrest?
```

Comment:
It's not forrest specific. It checks, whether "task" is a part of the protocol and if so get's its index, so we can have a look at the part right after it, since this is then likely the task label to return. so, if the actual protocol reads something like "whatever_task-rest_some_more" it would return "rest".
Whether that's useful for DTA, however, depends on whether or not that pattern shows up in its protocols.